clickhouse/client.h

```diff
@@ -47,8 +47,23 @@ struct ClientOptions {
         return *this; \
     }
 
+
+    /// List of hostnames with service ports
+    struct HostPort {
+        std::string host;
+        std::optional<unsigned int> port;
+
+        explicit HostPort(std::string host, std::optional<unsigned int> port = std::nullopt) : host(std::move(host)), port(std::move(port)) {
+        }
+    };
+    DECLARE_FIELD(hosts_ports, std::vector<HostPort>, SetHost,{});
     /// Hostname of the server.
-    DECLARE_FIELD(host, std::string, SetHost, std::string());
+    std::string host = std::string();
+    inline ClientOptions& SetHost(const std::string& value) {
+        hosts_ports.emplace_back(value, std::nullopt);
+        host = hosts_ports.back().host;
```

Comment:
Ok, so you keep host and HostPort. So host is duplicated. Can it be kept just one time?

Reply:
Yes, it can be kept just one time, I made relevant changes.